### PR TITLE
Spinner: add new prop labelPosition

### LINKED ISSUE #7228: Spinner: allow to customize label position
**Describe the feature that you would like added**
Add new prop `labelPosition` to allow rendering the label in different positions in regards to the `Spinner`

**What component or utility would this be added to**
`Spinner`

**Have you discussed this feature with our team, and if so, who**
Discussed with @betrue-final-final 

**Additional context/screenshots**
When doing some fluentization work, we discovered that the Fabric toolkit has all specs for the label on the side of the Spinner but none for what we actually have in the code, which is below the Spinner. Decided we need to add the option to customize this behavior with keeping the default to what we have currently in the master.

screen shot 2018-11-27 at 12 17 24 am

### PR DESCRIPTION
**Pull request by Vitalius1**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #7228
- [X] Include a change request file using `$ npm run change`

#### Description of changes

- Add new prop `labelPosition` to allow customizing what side of the `Spinner` to render the label
- Modify the styles to aid in the positioning of the label
- Add support for switching of the global classNames
- Add examples on the demo page
- Add snapshot test (none existed before)
- Add screener stories for all new cases with label position
- Remove an unused `interfaces.ts` file which was not exposed to our surface API or used anywhere else in the repo.

## Before:

screen shot 2018-11-27 at 12 31 57 am

## After:

screen shot 2018-11-27 at 12 31 33 am

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts [hunk 1]
```diff
@@ -78,6 +84,20 @@ export enum SpinnerSize {
   large = 3
 }
 
+export enum SpinnerLabelPosition {
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts]: I know we are moving away from using `const enum`. What about `enum`? Should I make this a `const` instead?

File: packages/office-ui-fabric-react/src/components/Spinner/interfaces.ts [hunk 2]
```diff
@@ -1,15 +0,0 @@
-/**
- * Deprecated at 2.0.0. Use `SpinnerSize` instead.
- * @deprecated Use `SpinnerSize` instead.
- */
-export enum SpinnerType {
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/Spinner/interfaces.ts]: Removing this entire file as we have these enums in the `Spinner.types.ts` file and the exported enums are not exposed to the surface API or used anywhere else in the repo

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts (hunk 1 above)
Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts]: Can we use a string union type here to remove the need for an enum? Example:

```ts
export type SpinnerLabelPosition = 'top' | 'right' | 'bottom' | 'left';
```

cc: @kenotron

File: packages/office-ui-fabric-react/src/components/Spinner/interfaces.ts (hunk 2 above)
Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/Spinner/interfaces.ts]: Seems reasonable. I confirmed this file is only accessible via absolute path import. It is _not_ exported at the component level nor are the enums used _at all_.

https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Spinner/index.ts

**Comment by KevinTCoughlin** [comment]
Looks like CI is currently failing because the API file needs to be updated via `npm run update-api`.

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts (hunk 1 above)
Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts]: I know @kenotron did replace `const enums` but I found that we are still using `enums` and I thought it's OK to use them. There are 2 more enums in the `Spinner.types.ts` file. What should I do regarding them to be consistent?

Comment (kenotron) [reply packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts]: enums are "okay", but we should really get rid of them and use string union types like @KevinTCoughlin suggested there.

Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts]: Wouldn't changing the current enums to string union types break everyone? I see that with the new ones we add the string union type, but with the existing ones, this would not work. Unless I am not understanding enums correctly.

**Comment by Vitalius1** [comment]
@cliffkoh 
screen shot 2018-11-27 at 4 11 47 pm
You mean that in RTL if the user says it wants the label on the right side it should be on the right side and not automaticaly switched by the css `flex-direction`?

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts (hunk 1 above)
Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts]: My understanding is:

For _existing_ enums, I would follow what was done in https://github.com/OfficeDev/office-ui-fabric-react/pull/7098/

But for _new_ enums, I would prefer string union types.

**Comment by cliffkoh** [comment]
@Vitalius1 actually you're right. I completely misread the example label - I thought it said the spinner should be on the right but it really says the Label is on the right..

Sorry for the confusion!

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts [hunk 3]
```diff
@@ -100,6 +112,7 @@ export interface ISpinnerStyleProps {
   theme: ITheme;
   size?: SpinnerSize;
   className?: string;
+  labelPosition?: SpinnerLabelPosition;
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts]: It would be great to document this.

Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/components/Spinner/Spinner.types.ts]: (p.s. really all of these should be documented but Spinner is one of the components that are a bad example :(...)

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.styles.tsx [hunk 4]
```diff
@@ -1,5 +1,11 @@
 import { ISpinnerStyleProps, ISpinnerStyles, SpinnerSize } from './Spinner.types';
-import { hiddenContentStyle, keyframes, HighContrastSelector } from '../../Styling';
+import { hiddenContentStyle, keyframes, HighContrastSelector, getGlobalClassNames } from '../../Styling';
+
+const GlobalClassNames = {
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Spinner/Spinner.styles.tsx]: Thanks for fixing this :+1:

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.test.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import * as renderer from 'react-test-renderer';
+
+import { Spinner } from './index';
+
+describe('Spinner', () => {
+  it('renders Spinner correctly', () => {
+    const component = renderer.create(<Spinner label="Standard spinner" />);
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Spinner/Spinner.test.tsx]: please add screener stories featuring the new label positions :)

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.test.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import * as renderer from 'react-test-renderer';
+
+import { Spinner } from './index';
+
+describe('Spinner', () => {
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Spinner/Spinner.test.tsx]: Thanks for writing spinner's very first unit test - it is a little odd it is missing the labelPosition but I guess the example tests cover it already.

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.styles.tsx (hunk 4 above)
Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/Spinner/Spinner.styles.tsx]: Seconded, awesome cleanup!

File: packages/office-ui-fabric-react/src/components/Spinner/Spinner.test.tsx (hunk 6 above)
Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/Spinner/Spinner.test.tsx]: Yea, I think the example tests are _good enough_ for `labelPosition`. A unit test could perhaps check that the default value is what we expect fairly easily, but its not a blocker IMO.

**Comment by Vitalius1** [comment]
@betrue-final-final Could you take a look and see if this is what we discussed this feature should be? Thanks!

**Comment by cliffkoh** [comment]
Concur with Kevin. Thanks for going the extra mile to clean things up and pursuing a high quality checkin. :+1:
